### DISCUSSION TITLE
Design tokens in Amazon Style Dictionary format

### DISCUSSION THREAD

**Discussion by VikingRock** [pattern-lab/patternlab-node#1425] [Ideas]
A lot of companies are using Amazon Style Dictionary to export their tokens from Sketch or Figma.
Patternlab token format is different. It will be super handy to support ASD token format.
https://amzn.github.io/style-dictionary/#/tokens

**Comment by mfranzke** [comment]
I really like your approach on this one as we were evaluating on using Style Dictionary as well and I would love to see both of these tools to integrate further, so thanks a lot for reminding me on this topic.

I've even already tried this out a little bit — in general from my understanding, there's not necessarily any incompatibility in case of the structure within the JSON files, as the included structured and expected schemas by Style Dictionary could get used to iterate through within pattern lab templates of any kind (I'm mainly using handlebars) now even already.
The main showstopper I could have identified is that I would love to define the pattern lab "_data" folder to provide this independently from the _pattern structure (like global data vs. pattern data), but the global data doesn't seem to be able to host those json files in a nested folder structure but only flat underneath the configured folder. So this would be something we would need to make more flexible.

Am I missunderstanding or missing an important aspect?

I'll try to provide my pattern lab & Amazon Style Dictionary setup asap to give some insight.
